R/output_analysis.R

```diff
@@ -14,3 +14,29 @@ growth_rate <- function(out) {
   lambdas <- pops / c(NA, pops[1:(length(pops) - 1)])
   exp(mean(log(lambdas), na.rm = TRUE))
 }
+
+
+#' Calculate annual growth rate
+#'
+#' @param out Model output data frame
+#' @returns Numeric vector of length one representing the annual factor by
+#' which the total tick population changes. To use this function, it is best
+#' to run the model for at least three years.
+#'
+#' @examples
+#' out <- run(config_ex_1)
+#' annual_growth_rate(out)
+#'
+#' @export
+annual_growth_rate <- function(out) {
+  daily_data <- data.frame(
+    day = unique(out$day)
+  )
+  daily_data$totalpop <- sapply(daily_data$day, function(x) sum(out$pop[out$day == x]))
+  daily_data$yr <- ceiling(daily_data$day / 365)
+  years <- unique(daily_data$yr)
+  maxpop <- sapply(years, function(x) max(daily_data$totalpop[daily_data$yr == x]))
+  ratechange <- numeric(length = length(maxpop) - 1)
+  for (i in seq_along(ratechange)) ratechange[i] <- maxpop[i + 1] / maxpop[i]
+  mean(ratechange[1:(length(ratechange) - 1)])
```

Comment:
This slicing drops the last element of the `ratechange` vector. When I ran this with 3 years of data, that meant that on this line we take the mean of 1 value -- I would have expected to take the mean of a vector of length two: (1) ratechange between years 1 and 2, and (2) ratechange between years 2 and 3. Is there a reason we don't include the last element?

Reply:
Yeah this was intentional. The idea behind this is incase the number of days does not result in a full year. That would mean the element of `maxpop` might not really be the maximum population in that year, it may come later. So I want to drop the last year. For example if there are 2000 days in the data set, the first five years are full (365*5 = 1825), but then the last year has only 175 days. So I don't want that one included.

This is a little hacky because if the number of days is a multiple of 365, then I am dropping the last year unnecessarily.

Reply:
If this makes sense to you, feel free to "resolve conversation". Or if you think we should fix so that it includes the last year if it is full, happy to do that too.

Reply:
That all makes sense to me, totally agree we shouldn't compare max population of an incomplete year. 

> fix so that it includes the last year if it is full

I think this would be great. I just an implementation for this fix in #56. Take a look and merge it into this branch if you like the approach/agree it's accomplishing what you describe

Reply:
@dallenmidd just to be clear, back over to you now

Reply:
That looks great! Thanks for that fix. I also noticed the example I had in the help function was giving an error because it didn't have two years. I changed the example and then merged that into this branch.. So I think we are all set with this branch too? You good to merge this into the master?

Reply:
Sweet! The only thing is that examples are run on CRAN to check the package, and they don't allow long-running code. Could you wrap the example (the two lines) with `dontrun{}` per [this](https://stackoverflow.com/a/12038225).

Reply:
Once that's done, all set to merge to `master`